testing/spectest/shared/common/light_client/single_merkle_proof.go

```diff
@@ -0,0 +1,195 @@
+	require.NoError(t, err)
+	beaconStateSSZ, err := snappy.Decode(nil, beaconStateFile)
+	require.NoError(t, err, "Failed to decompress")
+
+	var beaconState state.BeaconState
+	switch v {
+	case version.Altair:
+		beaconStateBase := &ethpb.BeaconStateAltair{}
+		require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+		beaconState, err = state_native.InitializeFromProtoAltair(beaconStateBase)
+		require.NoError(t, err)
+	case version.Bellatrix:
+		beaconStateBase := &ethpb.BeaconStateBellatrix{}
+		require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+		beaconState, err = state_native.InitializeFromProtoBellatrix(beaconStateBase)
+		require.NoError(t, err)
+	case version.Capella:
+		beaconStateBase := &ethpb.BeaconStateCapella{}
+		require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+		beaconState, err = state_native.InitializeFromProtoCapella(beaconStateBase)
+		require.NoError(t, err)
+	case version.Deneb:
+		beaconStateBase := &ethpb.BeaconStateDeneb{}
+		require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+		beaconState, err = state_native.InitializeFromProtoDeneb(beaconStateBase)
+		require.NoError(t, err)
+	case version.Electra:
+		beaconStateBase := &ethpb.BeaconStateElectra{}
+		require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+		beaconState, err = state_native.InitializeFromProtoElectra(beaconStateBase)
+		require.NoError(t, err)
+	default:
+		t.Fatalf("Unsupported version: %d", v)
+	}
+
+	beaconStateRoot, err := beaconState.HashTreeRoot(ctx)
+	require.NoError(t, err)
+	type Proof struct {
+		Leaf      string   `json:"leaf"`
+		LeafIndex uint64   `json:"leaf_index"`
+		Branch    []string `json:"branch"`
+	}
+	proofFile, err := util.BazelFileBytes(path.Join(testFolderPath, "proof.yaml"))
+	require.NoError(t, err)
+	var proof Proof
+	require.NoError(t, utils.UnmarshalYaml(proofFile, &proof))
+	leaf, err := hex.DecodeString(proof.Leaf[2:])
+	require.NoError(t, err)
+	require.NotNil(t, leaf)
+	var branch [][]byte
+	for _, b := range proof.Branch {
+		bBytes, err := hex.DecodeString(b[2:])
+		require.NoError(t, err)
+		branch = append(branch, bBytes)
+	}
+
+	var item []byte
+	if strings.Contains(testName, "current_sync_committee") {
+		syncCommittee, err := beaconState.CurrentSyncCommittee()
+		require.NoError(t, err)
+		item32, err := syncCommittee.HashTreeRoot()
+		require.NoError(t, err)
+		item = item32[:]
+	} else if strings.Contains(testName, "next_sync_committee") {
+		syncCommittee, err := beaconState.NextSyncCommittee()
+		require.NoError(t, err)
+		item32, err := syncCommittee.HashTreeRoot()
+		require.NoError(t, err)
+		item = item32[:]
+	} else if strings.Contains(testName, "finality_root") {
+		item = beaconState.FinalizedCheckpoint().Root
+	} else {
+		t.Skipf("Skipping test because of unknown test type BeaconState/%s", testName)
```

Comment:
```suggestion
		t.Fatalf("Skipping test because of unknown test type BeaconState/%s", testName)
```
This should be a hard failure. Folks should not be updating spectests without adding support for any new test type